pkg/store/store.go

```diff
@@ -39,10 +39,12 @@ import (
 	"sigs.k8s.io/yaml"
 
 	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
-	configurationv1 "github.com/kong/kubernetes-ingress-controller/pkg/apis/configuration/v1"
-	configurationv1beta1 "github.com/kong/kubernetes-ingress-controller/pkg/apis/configuration/v1beta1"
+	legacyv1 "github.com/kong/kubernetes-ingress-controller/pkg/apis/configuration/v1"
```

Comment:
-1 to naming this "legacy".

What if theoretically we never migrate off these? This is similar to comments like "I'll have time next month to fix this ugly bit" from 3 years ago 😂 

My suggestion is to keep these as they were, and have an issue in the backlog to move to the new apis.

Reply:
The `legacy` labeling in this case makes it clear these are the older API types in a codebase where we expect to have both for some time. I'm OK with changing the name, but I would prefer not to put these back the way they were as I want to have clarity. I'll think on some alternative nomenclature.

Reply:
After thinking on it for a while, it really seems as though legacy is appropriate as these are the predecessor APIs to the newly created ones and for a time we are going to have these duplicates.

I can't fathom the theoretical future where we don't migrate away from having duplicates and it seems that this is informative (it clearly defines that the "legacy" API are the previously existing ones). Is this a hard stop for you? Do you have any other naming conventions in mind?

Reply:
IMO legacy fits now that we have introduced the new implementations alongside them with the same versioning scheme. It's not so much a promise to move off them (though we do intend to) and more that we want something that's more descriptive at a glance than `configurationvX`, which didn't indicate of interest much to begin with. Now that we have two `configurationvX` packages, naming that indicates their relation to one another seems reasonable.